tools/eclipse-transformer-maven-extension/READEME.adoc

```diff
@@ -0,0 +1,186 @@
+            </extension>
+        </extensions>
+    </build>
+</project>
+----
+
+.Example Directory Tree After
+[source,text]
+----
+~/.m2/repository/io/smallrye/
+├── common
+│   ├── smallrye-common-annotation
+│   │   ├── 1.6.0
+│   │   │   ├── _remote.repositories
+│   │   │   ├── smallrye-common-annotation-1.6.0.jar
+│   │   │   ├── smallrye-common-annotation-1.6.0.jar.sha1
+│   │   │   ├── smallrye-common-annotation-1.6.0.pom
+│   │   │   └── smallrye-common-annotation-1.6.0.pom.sha1
+│   │   └── 1.6.0-$$jakarta9$$
+│   │       ├── smallrye-common-annotation-1.6.0-$$jakarta9$$.jar
+│   │       └── smallrye-common-annotation-1.6.0-$$jakarta9$$.pom
+│   ├── smallrye-common-bom
+│   │   ├── 1.6.0
+│   │   │   ├── _remote.repositories
+│   │   │   ├── smallrye-common-bom-1.6.0.pom
+│   │   │   └── smallrye-common-bom-1.6.0.pom.sha1
+│   │   └── 1.6.0-$$jakarta9$$
+│   │       └── smallrye-common-bom-1.6.0-$$jakarta9$$.pom
+│   ├── smallrye-common-classloader
+│   │   ├── 1.6.0
+│   │   │   ├── _remote.repositories
+│   │   │   ├── smallrye-common-classloader-1.6.0.jar
+│   │   │   ├── smallrye-common-classloader-1.6.0.jar.sha1
+│   │   │   ├── smallrye-common-classloader-1.6.0.pom
+│   │   │   └── smallrye-common-classloader-1.6.0.pom.sha1
+│   │   └── 1.6.0-$$jakarta9$$
+│   │       ├── smallrye-common-classloader-1.6.0-$$jakarta9$$.jar
+│   │       └── smallrye-common-classloader-1.6.0-$$jakarta9$$.pom
+│   ├── smallrye-common-constraint
+│   │   ├── 1.6.0
+│   │   │   ├── _remote.repositories
+│   │   │   ├── smallrye-common-constraint-1.6.0.jar
+│   │   │   ├── smallrye-common-constraint-1.6.0.jar.sha1
+│   │   │   ├── smallrye-common-constraint-1.6.0.pom
+│   │   │   └── smallrye-common-constraint-1.6.0.pom.sha1
+│   │   ├── 1.6.0-$$jakarta9$$
+│   │   │   ├── smallrye-common-constraint-1.6.0-$$jakarta9$$.jar
+│   │   │   └── smallrye-common-constraint-1.6.0-$$jakarta9$$.pom
+│   │   └── 1.6.0-$$jakarta9$$-$$jakarta9$$
+│   │       └── smallrye-common-constraint-1.6.0-$$jakarta9$$-$$jakarta9$$.pom
+│   ├── smallrye-common-expression
+│   │   ├── 1.6.0
+│   │   │   ├── _remote.repositories
+│   │   │   ├── smallrye-common-expression-1.6.0.jar
+│   │   │   ├── smallrye-common-expression-1.6.0.jar.sha1
+│   │   │   ├── smallrye-common-expression-1.6.0.pom
+│   │   │   └── smallrye-common-expression-1.6.0.pom.sha1
+│   │   └── 1.6.0-$$jakarta9$$
+│   │       ├── smallrye-common-expression-1.6.0-$$jakarta9$$.jar
+│   │       └── smallrye-common-expression-1.6.0-$$jakarta9$$.pom
+│   ├── smallrye-common-function
+│   │   ├── 1.6.0
+│   │   │   ├── _remote.repositories
+│   │   │   ├── smallrye-common-function-1.6.0.jar
+│   │   │   ├── smallrye-common-function-1.6.0.jar.sha1
+│   │   │   ├── smallrye-common-function-1.6.0.pom
+│   │   │   └── smallrye-common-function-1.6.0.pom.sha1
+│   │   └── 1.6.0-$$jakarta9$$-$$jakarta9$$
+│   │       ├── smallrye-common-function-1.6.0-$$jakarta9$$-$$jakarta9$$.jar
```

Comment:
Is the doubling of `-$$jakarta9$$` a typo or does that happen?

Reply:
I haven't been able to stop that from happening. It has to do with transitive dependencies I think, but I'm not sure where maven is pulling it from. Sometime I might have time to look again :)

Really a better solution will be to check the dependencies before this is invoked. However that is a lot more complicated.

Reply:
Will we see an extra `$$jakarta9$$` per transitive dependency level?  For example, do the transitive dependencies for smallrye-common have `$$jakarta9$$-$$jakarta9$$-$$jakarta9$$`?

Reply:
Not that I've seen. There seems to only be cases of a double suffix for some reason. It's very strange.

FWIW we may not end up needing this in RESTEasy if Infinispan releases a namespace change. We're moving Spring and MicroProfile stuff out of RESTEasy into their own projects. I can convert this to a draft if you'd prefer to ensure it doesn't get merged unless it's needed or until I can fix the double suffix.

Reply:
Thanks, converting to draft sounds like a good approach for now.  If we later decide that we need this and that users can work around the double suffix thing for transitive dependencies, we could follow that approach.